packages/cozy-client/src/devtools/Pouch.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useMemo, useState, useCallback, useEffect } from 'react'
+import CozyPouchLink from 'cozy-pouch-link'
+import Button from 'cozy-ui/transpiled/react/MuiCozyTheme/Buttons'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import format from 'date-fns/format'
+
+import useClient from '../hooks/useClient'
+import PanelContent from './PanelContent'
+
+const listStyle = { paddingLeft: '1rem' }
+
+/**
+ * Allows to view state and manage the PouchLink of the current cozy
+ * client.
+ */
+const PouchDevTool = () => {
+  const client = useClient()
+  const [status, setStatus] = useState('NA')
+  const pouchLink = useMemo(() => {
+    return client.links.find(link => link instanceof CozyPouchLink)
+  }, [client])
+  const [resetting, setResetting] = useState(false)
+  useEffect(() => {
+    const handleReplicationStart = () => setStatus('replicating')
+    const handleReplicationStop = () => setStatus('stopped')
+    const handleReplicationEnd = () => setStatus('idle')
+    client.on('pouchlink:sync:start', handleReplicationStart)
+    client.on('pouchlink:sync:stop', handleReplicationStop)
+    client.on('pouchlink:sync:end', handleReplicationEnd)
+    return () => {
+      client.removeListener('pouchlink:sync:start', handleReplicationStart)
+      client.removeListener('pouchlink:sync:stop', handleReplicationStop)
+      client.removeListener('pouchlink:sync:end', handleReplicationEnd)
+    }
+  }, [client])
+
+  const handleClickStartReplication = useCallback(async () => {
+    // Need to stop the replication loop before restarting it
+    // otherwise startReplication has no effect since the replication
+    // loop is already started.
+    pouchLink.stopReplication()
+    pouchLink.startReplication()
+  }, [pouchLink])
+
+  const handleClickStopReplication = useCallback(async () => {
+    pouchLink.stopReplication()
+  }, [pouchLink])
+
+  const handleClickReset = useCallback(async () => {
+    setResetting(true)
+
+    try {
+      await pouchLink.reset()
+      if (window.confirm('Pouches have been destroyed, reload the page ?')) {
+        window.location.reload()
+      }
+    } catch (e) {
+      setResetting(false)
+    }
+  }, [setResetting, pouchLink])
+
+  if (!pouchLink) {
+    return (
+      <PanelContent>
+        <p>Pouch is not active</p>
+      </PanelContent>
+    )
+  }
+  return (
+    <PanelContent>
+      <Typography variant="subtitle1">Doctypes saved in Pouch</Typography>
+      <ul style={listStyle}>
+        {pouchLink.doctypes.map(doctype => (
+          <li key={doctype}>{doctype}</li>
+        ))}
+      </ul>
+      <Typography variant="subtitle1">Last replication</Typography>
+      <p>
+        {pouchLink.lastReplicatedAt
+          ? format(pouchLink.lastReplicatedAt, 'HH:mm:ss DD/MM/YYYY')
+          : 'NA'}
+      </p>
```

Comment:
[troll on: instead of storing this in a new var, why not using the last sync attribute? https://docs.cozy.io/en/cozy-stack/settings/#post-settingssynchronized It'll force us to post the information troll off]. It'll work only for OAuth client... :( 


To be more productive, I don't know if this is the real lastReplicatedAt. If the sync was already done, we refresh the app, the lastReplicatedAt will be NA even though the replication was already done. Can't we get this information from Pouch itself?

Reply:
This panel is meant to only be a view on the pouch link. I think the synchronisation status to the cozy is bit different and I am not quite aware of the semantic differences :/

As for the global replication status on the pouch link, in fact I doubt that this is useful since the link executes the query based on the per-doctype synchronisation status.

I've modified the PR to have synchronisation statuses per doctype. It allowed me also to piggyback on the syncedDoctypes attributes of the pouch manager that is automatically saved in localStorage so that it works across reloads.

<img width="417" alt="image" src="https://user-images.githubusercontent.com/465582/129038638-bcad0d92-a3e6-46b0-8aae-190a00ba61ad.png">